server/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -141,144 +144,150 @@ static boolean isIdleThread(ThreadInfo threadInfo) {
         return false;
     }
 
-    String innerDetect(ThreadMXBean threadBean, long currentThreadId) throws Exception {
-        if (threadBean.isThreadCpuTimeSupported() == false) {
-            throw new ElasticsearchException("thread CPU time is not supported on this JDK");
-        }
+    Map<Long, ThreadTimeAccumulator> getAllValidThreadInfos(ThreadMXBean threadBean, long currentThreadId) {
```

Comment:
By subtracting the before timings from the after timings, we are able to reuse the thread timing capture code and catch timings of threads that had started during the 'profiling' (sleeping) time interval.